src/service-module/make-base-model.ts

```diff
@@ -38,48 +47,51 @@ export default function makeBaseModel(options: FeathersVuexOptions) {
   const addModel = prepareAddModel(options)
   const { serverAlias } = options
 
-  // If this serverAlias already has a BaseModel, nreturn it
+  // If this serverAlias already has a BaseModel, return it
   const ExistingBaseModel = _get(globalModels, `[${serverAlias}].BaseModel`)
   if (ExistingBaseModel) {
-    return ExistingBaseModel
+    return ExistingBaseModel as ModelStatic
   }
 
-  abstract class BaseModel {
+  abstract class BaseModel implements Model {
     // Think of these as abstract static properties
     public static servicePath: string
     public static namespace: string
     public static keepCopiesInStore = options.keepCopiesInStore
     // eslint-disable-next-line
-    public static instanceDefaults(data, { models, store }) {
+    public static instanceDefaults(data: AnyData, ctx: ModelSetupContext) {
       return data
     }
     // eslint-disable-next-line
-    public static setupInstance(data, { models, store }) {
+    public static setupInstance(data: AnyData, ctx: ModelSetupContext) {
       return data
     }
-    public static diffOnPatch(data) {
+    public static diffOnPatch(data: AnyData) {
       return data
     }
 
     // Monkey patched onto the Model class in `makeServicePlugin()`
-    public static store: Record<string, any>
+    public static store: Store<StoreState>
 
     public static idField: string = options.idField
     public static tempIdField: string = options.tempIdField
     public static preferUpdate: boolean = options.preferUpdate
     public static serverAlias: string = options.serverAlias
 
-    public static readonly models = globalModels // Can access other Models here
-    public static copiesById = {}
+    public static readonly models = globalModels as GlobalModels // Can access other Models here
+    public static copiesById: {
+      [key: string]: Model | undefined
+      [key: number]: Model | undefined
+    } = {}
 
     public __id: string
     public __isClone: boolean
-    public data: Record<string, any>
```

Comment:
As far as I can tell, data was not being used anywhere so I removed it. I thought perhaps it was a remnant from a previous implementation? Can add back if necessary.